src/main/java/budgetbuddy/model/AccountsManager.java

```diff
@@ -19,8 +19,8 @@
  * Manages the accounts in a list of accounts.
  */
 public class AccountsManager {
-    private final FilteredList<Account> filteredAccounts;
     private final UniqueAccountList accounts;
+    private FilteredList<Account> filteredAccounts;
```

Comment:
not sure why you removed the final declaration here. I don't think setting the predicate will change the reference of the list soo...